src/Microsoft.Diagnostics.Monitoring.StartupHook/Exceptions/Eventing/ExceptionsEventSource.cs

```diff
@@ -167,13 +202,51 @@ private unsafe void WriteEventCore(int eventId, Span<EventData> data)
             data.Size = sizeof(T);
         }
 
+        [NonEvent]
+        private static unsafe void SetValue(ref EventData data, in Span<byte> value)
+        {
+            data.DataPointer = (nint)Unsafe.AsPointer(ref value.GetPinnableReference());
```

Comment:
Can you put some comments here regarding the expected type of input? For example, if you forget to pin reference data and send it here as a span, a GC could destroy the validity of the pointer since it's not actually pinned.

Reply:
Done